components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,158 @@
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @return An initialized MDCActionSheetController object.
+ */
++ (nonnull instancetype)actionSheetControllerWithTitle:(nullable NSString *)title;
+
+/**
+ Convenience constructor to create and return a view controller for displaying an alert to the user.
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @param message Descriptive text that summarizes a decision in a sentence of two.
+ @return An initialized MDCActionSheetController object.
+ */
++ (nonnull instancetype)actionSheetControllerWithTitle:(nullable NSString *)title
+                                               message:(nullable NSString *)message;
+
+- (nonnull instancetype)initWithNibName:(NSString *)nibNameOrNil
+                                 bundle:(NSBundle *)nibBundleOrNil;
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder;
+
+/**
+ Adds an action to the action sheet.
+
+ Actions are the possible reactions of the user to the presented alert. Actions are added as a
+ list item in the list of the action sheet.
+ Action buttons will be laid out from right to top to bottom depending on the the order they
+ were added, first being top last being bottom.
+
+ @param action Will be added to the end of MDCActionSheetController.actions.
+ */
+- (void)addAction:(nonnull MDCActionSheetAction *)action;
+
+/**
+ The actions that the user can take in response to the action sheet.
+
+ The order of the actions in the array matches the order in which they were added to the action sheet.
+ */
+@property (nonatomic, nonnull, readonly) NSArray<MDCActionSheetAction *> *actions;
+
+/**
+ The title of the action sheet controller.
+ */
+@property (nonatomic, nullable, copy) NSString *title;
+
+/**
+ The message of the action sheet controller.
+ */
+@property (nonatomic, nullable, copy) NSString *message;
+
+/*
+ Indicates whether the button should automatically update its font when the device’s
+ UIContentSizeCategory is changed.
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIConnectSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ If set to YES, this button will base its text font on MDCFontTextStyleButton.
+
+ Defaults value is NO.
+ */
+@property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
+BOOL mdc_adjustsFontForContentSizeCategory;
+
+@end
+
+/**
+ MDCActionHandler is a block that will be invoked when the action is selected.
+ */
+typedef void (^MDCActionSheetHandler)(MDCActionSheetAction *_Nonnull action);
+
+/**
+  MDCActionSheetAction is passed to MDCActionSheetController to add a button to the action sheet.
+ */
+@interface MDCActionSheetAction : NSObject <NSCopying>
+
+
+/**
+  Action alerts control the list items that will be displayed on the bottom of an action
```

Comment:
Not sure if this sentence is as-intended? Did you mean something like "Returns an action sheet action populated with the given values"?

Reply:
Done